src/main/kotlin/compiler/ast/Node.kt

```diff
@@ -31,8 +31,17 @@ abstract class Node {
     // Identify the type of any identifiers we point to.
     open fun identify() { }
 
+    // Collect all variable names under this node.
```

Comment:
We use this new Node.collectVars() to find all referenced vars under the N_BLOCK of our lambda, so we can capture from  the enclosing scope.